platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/DefaultListProperty.java

```diff
@@ -81,4 +86,29 @@ public ListProperty<T> convention(Provider<? extends Iterable<? extends T>> prov
         super.convention(provider);
         return this;
     }
+
+    @Override
+    public HasMultipleValues<T> withActualValue(Action<CollectionPropertyConfigurer<T>> action) {
```

Comment:
:nail_care: shouldn't all builder-style properties return the narrower type (like `ListProperty` here, and `SetProperty` in `DefaultSetProperty`), for users convenience?

Reply:
Added covariant overrides